py/private/py_binary.bzl

```diff
@@ -150,17 +185,33 @@ _attrs = dict({
     "_interpreter_version_flag": attr.label(
         default = "//py:interpreter_version",
     ),
+    # Required for py_version attribute
+    "_allowlist_function_transition": attr.label(
+        default = "@bazel_tools//tools/allowlists/function_transition_allowlist",
+    ),
 })
 
 _attrs.update(**_py_library.attrs)
 
+def _python_version_transition_impl(_, attr):
+    if not attr.python_version:
+        return {}
+    return {"@rules_python//python/config_settings:python_version": str(attr.python_version)}
+
+_python_version_transition = transition(
+    implementation = _python_version_transition_impl,
+    inputs = [],
+    outputs = ["@rules_python//python/config_settings:python_version"],
```

Comment:
I dont see this https://github.com/bazelbuild/rules_python/blob/main/python/config_settings/BUILD.bazel - does that mean we need a change to rules_python to land and get released first?

Reply:
it is created by this macro, https://github.com/bazelbuild/rules_python/blob/49cdf7d3fe000076d6432a34238e5d25f5b598d0/python/config_settings/BUILD.bazel#L28-L30

https://github.com/bazelbuild/rules_python/blob/49cdf7d3fe000076d6432a34238e5d25f5b598d0/python/private/config_settings.bzl#L173-L182

Reply:
We did not need to introduce anything new to rules_python, everything works with existing rules_python version.

Reply:
oh, neat, they were so close to getting this right :)